src/components/DataTable.js

```diff
@@ -52,317 +89,256 @@ export default Utils.withCancellationSignal(class DataTable extends Component {
-                columns: [
-                  ...(selectionModel ? [{
-                    width: 70,
-                    headerRenderer: selectionModel.type === 'multiple' ? () => {
-                      return h(Fragment, [
-                        h(Checkbox, {
-                          checked: this.pageSelected(),
-                          disabled: !entities.length,
-                          onChange: () => this.pageSelected() ? this.deselectPage() : this.selectPage()
-                        }),
-                        h(PopupTrigger, {
-                          closeOnClick: true,
-                          content: h(Fragment, [
-                            h(MenuButton, { onClick: () => this.selectPage() }, ['Page']),
-                            h(MenuButton, { onClick: () => this.selectAll() }, [`All (${totalRowCount})`]),
-                            h(MenuButton, { onClick: () => this.selectNone() }, ['None'])
-                          ]),
-                          side: 'bottom'
-                        }, [
-                          h(Clickable, { 'aria-label': '"Select All" options' }, [icon('caretDown')])
-                        ])
-                      ])
-                    } : () => div(),
-                    cellRenderer: ({ rowIndex }) => {
-                      const thisEntity = entities[rowIndex]
-                      const { name } = thisEntity
-                      const { type } = selectionModel
+  useEffect(() => {
+    loadData()
+    if (persist) {
+      StateHistory.update({ itemsPerPage, pageNumber, sort, activeTextFilter })
+    }
+  }, [itemsPerPage, pageNumber, sort, activeTextFilter, refreshKey]) // eslint-disable-line react-hooks/exhaustive-deps
 
-                      if (type === 'multiple') {
-                        const { selected, setSelected } = selectionModel
-                        const checked = _.has([name], selected)
-                        return h(Checkbox, {
-                          'aria-label': name,
-                          checked,
-                          onChange: () => setSelected((checked ? _.unset([name]) : _.set([name], thisEntity))(selected))
-                        })
-                      } else if (type === 'single') {
-                        const { selected, setSelected } = selectionModel
-                        return h(RadioButton, {
-                          'aria-label': name,
-                          name: 'entity-selection',
-                          checked: _.isEqual(selected, thisEntity),
-                          onChange: () => setSelected(thisEntity)
-                        })
-                      }
+  useEffect(() => {
+    if (persist) {
+      setLocalPref(persistenceId, { columnWidths, columnState })
+    }
+  }, [columnWidths, columnState]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    table.current?.recomputeColumnSizes() // eslint-disable-line no-unused-expressions
```

Comment:
Perhaps the thing to do is to make the `GridTable` recompute column sizes whenever the `columns` prop changes? That way we wouldn't have to expose it externally at all.

Reply:
Also, I think if we convert hook in the `GridTable` to a layout effect, we wouldn't have to call the internal `ScrollSync` private method...?

Reply:
In order for `columns` to be the trigger, we'd need to memoize it to establish a stable reference, and its dependencies are a superset of what's required. Notably, it depends on the list of rows, which doesn't affect the column sizes. We could split the column width data out into a separate prop, but that would make the API less nice.

TBH I don't remember why we're calling `_onScroll`. It looks like it may be trying to initialize the `ScrollSync` internal state based on the saved position. But it doesn't look like the mechanism to save scroll positions is working correctly anyway.

Reply:
I fixed the scroll restoration, take a look.